crowbar_framework/lib/crowbar/backup/restore.rb

```diff
@@ -296,8 +297,12 @@ def proposal?(file)
         raw_data.key?("attributes") && raw_data.key?("deployment")
       end
 
-      def migrate_database(time)
-        Crowbar::Migrate.migrate!
+      def migrate_database(time, migration_level = nil)
+        if migration_level
+          Crowbar::Migrate.migrate_to(migration_level)
```

Comment:
back to the question 

> What happens if that migration level is not known yet? (ie: use new backup on old code)

I think I we were not on the same page, I thought you mean the migration_level attribute, or were you talking about the migration_level in the db?

if it is about the attribute: 
wouldn't it just be ignored then? it depends on how old the code is I guess. but if it is at the code state of release, then the migration just runs two times without the migration_level distinction.

if it is about the database entry: 
we could just add a check if that migration exists in `migrator.get_all_versions` and when it doesn't exist we return and show a message in the UI.
would that be fine with you?